autoreject/autoreject.py

```diff
@@ -767,24 +846,203 @@ def transform(self, epochs, return_log=False):
+    
+    # GPU interpolation - returns tensor
+    data_gpu = gpu_interpolate_bad_epochs(
+        data, interp_ch_indices, picks, pos, device=device
+    )
+    
+    # Copy back to epochs (only picked channels were interpolated)
+    epochs._data = data_gpu.cpu().numpy()
+
+
 def _interpolate_bad_epochs(
-        epochs, interp_channels, picks, dots=None, verbose=True):
-    """Actually do the interpolation."""
+        epochs, interp_channels, picks, dots=None, verbose=True, n_jobs=1, 
+        use_gpu=False, device=None):
+    """Actually do the interpolation.
+    
+    Parameters
+    ----------
+    epochs : mne.Epochs
+        The epochs to interpolate (modified in-place).
+    interp_channels : list of list of str
+        For each epoch, list of channel names to interpolate.
+    picks : array-like
+        Channel indices to consider.
+    dots : tuple | None
+        Precomputed dots for interpolation.
+    verbose : bool
+        Whether to show progress bar.
+    n_jobs : int
+        Number of parallel jobs. Default 1 (sequential).
+    use_gpu : bool
+        If True, use GPU-accelerated interpolation.
+    device : str or None
+        GPU device to use when use_gpu=True.
+    """
     assert len(epochs) == len(interp_channels)
+    
+    # Try GPU path if requested
+    if use_gpu:
+        try:
+            _interpolate_bad_epochs_gpu(epochs, interp_channels, picks, 
+                                        device=device, verbose=verbose)
+            return
+        except Exception as e:
+            if verbose:
+                print(f"GPU interpolation failed, falling back to CPU: {e}")
+    
     pos = 2
-
-    for epoch_idx, interp_chs in _pbar(
-            list(enumerate(interp_channels)),
-            desc='Repairing epochs',
-            position=pos, leave=True, verbose=verbose):
-        epoch = epochs[epoch_idx]
-        epoch.info['bads'] = interp_chs
-        interpolate_bads(epoch, dots=dots, picks=picks, reset_bads=True)
-        epochs._data[epoch_idx] = epoch._data
+    
+    # Check if we can parallelize (n_jobs > 1 and enough epochs)
+    if n_jobs != 1 and len(epochs) > 1:
+        # Parallel execution
+        parallel = Parallel(n_jobs=n_jobs, verbose=0)
+        my_interp = delayed(_interpolate_single_epoch)
+        
+        results = parallel(
+            my_interp(
+                epochs._data[epoch_idx].copy(),
+                epochs.info,
+                interp_chs,
+                dots,
+                picks
+            )
+            for epoch_idx, interp_chs in _pbar(
+                list(enumerate(interp_channels)),
+                desc='Repairing epochs (parallel)',
+                position=pos, leave=True, verbose=verbose
+            )
+        )
+        
+        # Update epochs data with interpolated results
+        for epoch_idx, interpolated_data in enumerate(results):
+            epochs._data[epoch_idx] = interpolated_data
+    else:
```

Comment:
do you need the if/else? Doesn't joblib fallback to the original behavior if n_jobs = 1?